src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,210 @@
+            if (Integer.parseInt(line.split(key)[1].trim()) > task.getTaskCount()) {
+                return true;
+            }
+        }
+        return (line.length() <= FIVE);
+    }
+    private static void createSave() {
+        File saveFile = new File(Duke.fileName);
+        File saveDir = new File(Duke.fileDir);
+        if (saveDir.mkdirs()) {
+            System.out.println("Successfully created save dir");
+        } else {
+            System.out.println("Save folder already exists.");
+        }
+        try {
+            if (saveFile.createNewFile()) {
+                System.out.println("Successfully created save file");
+            } else {
+                System.out.println("Save file already exists");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void updateSave() {
+        Task[] saveLists = new Task[items.size()];
+        items.toArray(saveLists);
+        try {
+           FileWriter fw = new FileWriter(Duke.fileName);
+            for (Task saveList : saveLists) {
+                fw.write(saveList.toString().charAt(1) + "/");
+                fw.write(saveList.getStatus() + "/" + saveList.getDescription());
+                if (!saveList.getDate().equals("empty"))  {
+                    fw.write("/" + saveList.getDate());
+                }
+                fw.write("\n");
+            }
+            System.out.println("Automatically saved to " + fileName);
+            fw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void readSave(CommandList task) throws FileNotFoundException {
+        File saveFile = new File(Duke.fileName);
+        Scanner s = new Scanner(saveFile);
+        while (s.hasNext()) {
+            String[] chars = s.nextLine().split("/");
+            String type = chars[0].trim();
+            String status = chars[1].trim();
+            String description = chars[2].trim();
+            switch (type) {
+                case "E":
+                    String date = chars[3].trim();
+                    task.addEvent(Duke.items, description, date);
+                    if (status.equals("1")) {
+                        items.get(task.getTaskCount()).markDone();
+                    }
+                    break;
+                case "D":
+                    date = chars[3].trim();
+                    task.addDeadline(Duke.items, description, date);
+                    if (status.equals("1")) {
+                        items.get(task.getTaskCount()).markDone();
+                    }
+                    break;
+                case "T":
+                    task.addTodo(Duke.items, description);
+                    break;
+            }
+        }
+        task.loadTaskCount(items);
+        System.out.println("Save file successfully loaded");
+    }
+
+    public static String readCommand(Scanner in, CommandList task, DukeException error) {
```

Comment:
This method is parsing the command string, but it is not obvious how it is done with seemingly arbitrary constants like` FOUR FIVE EIGHT` etc. You could consider a more direct way of comparing strings and checking parameters.